plugins/vpp/ifplugin/vppcalls/vpp2009/wireguard_vppcalls.go

```diff
@@ -0,0 +1,119 @@
+//  Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package vpp2009
+
+import (
+	"encoding/base64"
+	"fmt"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/binapi/vpp2009/interface_types"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/binapi/vpp2009/ip_types"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/binapi/vpp2009/wireguard"
+	"go.ligato.io/vpp-agent/v3/plugins/vpp/ifplugin/vppcalls"
+	interfaces "go.ligato.io/vpp-agent/v3/proto/ligato/vpp/interfaces"
+	"net"
+)
+
+// AddWireguardTunnel adds new Wireguard tunnel interface.
+func (h *InterfaceVppHandler) AddWireguardTunnel(ifName string, wireguardLink *interfaces.WireguardLink) (uint32, error) {
```

Comment:
Add checks if wireguard rpc client is nil to avoid sending requests in case the wireguard plugin is disabled. Similar how wmxnet3 interface is done: https://github.com/glazychev-art/vpp-agent/blob/wireguard_plugin/plugins/vpp/ifplugin/vppcalls/vpp2009/vmxnet3_vppcalls.go#L30-L32